main.tf

```diff
@@ -1,45 +1,56 @@
 provider "triton" {
-    url = "https://${var.dc_name}.api.joyent.com"
-    account = "${var.triton_account}"
-    key_id = "${var.triton_key_id}"
+    # Commenting these out because they will take the SDC_URL, SDC_ACCOUNT, and SDC_KEY_ID env vars as defaults
+    # url = "https://${var.dc_name}.api.joyent.com"
+    # account = "${var.triton_account}"
+    # key_id = "${var.triton_key_id}"
 }
 
+#
+# Details about all deployments of this application
+#
 data "triton_network" "public" {
     name = "Joyent-SDC-Public"
 }
 
-data "triton_image" "green_image" {
-    name = "${var.green_image_name}"
-    version = "${var.green_image_version}"
+#
+# Details about the "blue" deployment
+#
+data "triton_image" "blue_image" {
+    name = "${var.blue_image_name}"
+    version = "${var.blue_image_version}"
     type = "lx-dataset"
     most_recent = true
 }
 
-resource "triton_machine" "green_machine" {
```

Comment:
The count of green instances is zero, so I moved green to the bottom of the files and elevated blue.